src/go/k8s/pkg/resources/resource.go

```diff
@@ -30,5 +31,5 @@ type Resource interface {
 	Kind() string
 
 	// Ensure reconcile only one resource available in Kubernetes API server
-	Ensure(ctx context.Context) error
+	Ensure(ctx context.Context, logger logr.Logger) error
```

Comment:
Glad to see we're making a logger available. What do you think of the alternative of adding it as a field under the resource? Basically, our resources act as reconcilers. I've seen operators pass a logger instance around while others make it part of each reconciler struct (like our Cluster reconciler does). With deeper chains of calls we'll have to be adding "logger" as an argument in various functions. That's my only concern. What do you think, pros/cons?

Reply:
oh yeah, I thought about it but since ensure is the only place that does anything meaningful I thought it would be OK to inject it like this. But yeah, it's true we'll need to pass it around, I'll move it to the resource. Thanks

Reply:
btw. this moved to #584 and I've implemented the change you proposed